vividus-bdd-engine/src/main/java/org/vividus/bdd/transformer/FilteringTableTransformer.java

```diff
@@ -98,6 +85,24 @@ public String transform(String tableAsString, ExamplesTableProperties properties
                 .orElseGet(examplesTable::getRows);
     }
 
+    private List<String> getFilteredHeaders(String byMaxColumns, String byColumnNames, List<String> headerValues)
+    {
+        int columnsLimit = byMaxColumns == null
+                ? headerValues.size() : Math.min(headerValues.size(), Integer.parseInt(byMaxColumns));
+        List<String> filteredColumns = headerValues;
```

Comment:
`List<String> filteredColumns = new ArrayList<>(headerValues);` - copy is required to avoid changes in the original input